.circleci/config.yml

```diff
@@ -1,55 +1,155 @@
 version: 2.1
 
 orbs:
+  github-cli: circleci/github-cli@1.0
   tools: replikativ/clj-tools@0
 
+jobs:
+  setup:
+    executor: tools/clojurecli
+    parameters:
+      setup_cljs:
+        type: boolean
+        default: false
+    steps:
+      - restore_cache:
+          keys:
+            - source-{{ .Branch }}-{{ .Revision }}
+            - source-{{ .Branch }}
+            - source-
+      - checkout
+      - save_cache:
+          key: source-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - .git
+      - restore_cache:
+          keys:
+            - deps-{{ checksum "deps.edn" }}
+            - deps-
+      - run:
+          name: resolve deps
+          command: clojure -P
+      - save_cache:
+          key: deps-{{ checksum "deps.edn" }}
+          paths:
+            - /home/circleci/.m2
+      - when:
+          condition: <<parameters.setup_cljs>>
+          steps:
+            - restore_cache:
+                keys:
+                  - npm-{{ checksum "package-lock.json" }}
+                  - npm-
+            - run: npm ci
+            - save_cache:
+                key: npm-{{ checksum "package-lock.json" }}
+                paths:
+                  - /home/circleci/.npm
+      - persist_to_workspace:
+          root: /home/circleci/
+          paths:
+            - .m2
+            - .npm
+            - replikativ
+  build:
+    executor: tools/clojurecli
+    parameters:
+      build_cljs:
+        type: boolean
+        default: false
+    steps:
+      - attach_workspace:
+          at: /home/circleci
+      - run:
+          name: clean
+          command: clojure -Sthreads 1 -T:build clean
+      - run:
+          name: jar
+          command: clojure -Sthreads 1 -T:build jar
+      - when:
+          condition: <<parameters.build_cljs>>
+          steps:
+            - run:
+                name: compile_cljs
+                command: npx shadow-cljs compile ci
+                no_output_timeout: 5m
+      - persist_to_workspace:
+          root: /home/circleci/
+          paths:
+            - .m2
+            - replikativ
+
+  test:
+    executor: tools/clojurecli
+    parameters:
+      test_cljs:
+        type: boolean
+        default: false
+    steps:
+      - attach_workspace:
+          at: /home/circleci
+      - run:
+          name: test
+          command: clojure -Sthreads 1 -T:build test
```

Comment:
I would run `./bin/run-unittests` here for now since you're using still kaocha as a test dependency.

Reply:
ok